### ISSUE TITLE
Do not include source files and/or name include files properly

### ISSUE THREAD

**Issue by dvrogozh** [intel/gmmlib#32]
https://github.com/intel/gmmlib/blob/3bbe7417630860d84d5d28799fc7226dd1c5b094/Source/GmmLib/inc/External/Common/GmmCommonExt.h#L63

Please, do not include source files and/or name include files properly. Other places may exist.

**Comment by drprajap** [comment]
This source file is a special inclusion which works as .c and header file as well depending on purpose...it is included by design and is intentional.

**Comment by chivakker** [comment]
@drprajap can the relative directory #includes be fixed as well?  If your cmake is already including -I 'pkg-config -cflags igdgmm', then all includes should relate on that absolute directory and includes should look like:

#include "Utility/CpuSwizzleBit/CPUswizzleBit.h"

c files should not be included.  If that's your design, please review your design.

**Comment by drprajap** [comment]
Yes, relative directory path can be fixed and I was checking the builds with that change. when I use 
#include "Utility/CpuSwizzleBit/CPUswizzleBit.h"
with igdgmm package present on the system, media-driver builds fine. but When the package is not present and it builds from gmmlib sources, the media-driver compilation fails with "file not found" error.

I am looking into the issue right now and will update on that.

Regarding other includes, we can fix all other hard coded paths with this change. But reorganizing headers requires bigger effort, which cannot be taken at this point, because team is working on other higher priority tasks. Once all the clients transition is successful with DLL, We will take the reorganization with priority along with API cleanup. That is next task in our plate.

Regarding c file inclusion, as per design, CpuSwizzleBlt is standalone feature and we wanted to keep it independent so that it can be used in other projects/tools without any dependency. If we divide in 2 headers, we add additional dependency for other projects and it easily gets polluted with other non-relevant stuff. This file rarely changes so we do not want to consider change to the design.

**Comment by uartie** [comment]
Please fix this asap... media-driver won't compile when using gmmlib from pkg-config settings.

```
<include-prefix>/igdgmm/inc/External/Common/GmmCommonExt.h:63:10: fatal error: ../../../Utility/CpuSwizzleBlt/CpuSwizzleBlt.c: No such file or directory
 #include "../../../Utility/CpuSwizzleBlt/CpuSwizzleBlt.c"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Clearly, I see ```<include-prefix>/igdgmm/GmmLib/Utility/CpuSwizzleBlt/CpuSwizzleBlt.c``` is installed, but the relative include in ```GmmCommonExt.h``` is completely wrong.

```
gmmlib (master) tags/intel-gmmlib-18.3.pre2-0-g5b61c8a6b188 
media-driver (master) heads/master-0-g6d80b29459b8
```

**Comment by dvrogozh** [comment]
@uartie: I am using gmmlib from pkg-config, but I don't see this issue. I am not CentOS 7.5. This issue looks to be compiler/OS specific. Maybe you can provide a PR fix for that since you can try it?

**Comment by uartie** [comment]
@dvrogozh please make sure you don't have a stale installation of gmmlib (i.e. remove your old <install-prefix>/igdgmm directory before reinstalling with latest code).  Let me know if you still don't see this after that.  IIRC, I didn't see this problem last week.

**Comment by uartie** [comment]
I wouldn't suspect install would be compiler/os specific other than the base install prefix

**Comment by uartie** [comment]
> @dvrogozh please make sure you don't have a stale installation of gmmlib (i.e. remove your old <install-prefix>/igdgmm directory before reinstalling with latest code). Let me know if you still don't see this after that. IIRC, I didn't see this problem last week.

My apologies, turns out I had a stale install of gmmlib, doh (foot in mouth)!  Error is gone for me now.

**Comment by dvrogozh** [comment]
@nehudesi : you build error don't seem to be related with gmmlib at all. Instead this seem to be related with mismatch libva and media-driver. Please, make sure that you checked out the following tags, build and install dependencies in the following order:
* https://github.com/intel/gmmlib/releases/tag/intel-gmmlib-18.3.0
* https://github.com/intel/libva/releases/tag/2.3.0
* https://github.com/intel/libva-utils/releases/tag/2.3.0
* https://github.com/intel/media-driver/releases/tag/intel-media-18.3.0
If issue persist - please, file issue on media-driver side.

**Comment by nehudesi** [comment]
Appreciate your reply @dvrogozh ! I will work on that and let you know!
